nats_events/nats_events/v1_0/nats_queue/events/__init__.py

```diff
@@ -0,0 +1,182 @@
+    """Called on shutdown."""
+    LOGGER.info("NATS plugin shutdown hook called.")
+    nats = profile.inject_or(NATS)
+    if nats:
+        await nats.close()
+        LOGGER.info("NATS connection closed.")
+
+
+def _derive_category(topic: str):
+    match = RECORD_RE.match(topic)
+    if match:
+        return match.group(1)
+    if WEBHOOK_RE.match(topic):
+        return "webhook"
+
+
+def process_event_payload(event_payload: Any):
+    """Process event payload."""
+    processed_event_payload = None
+    if isinstance(event_payload, dict):
+        processed_event_payload = event_payload
+    else:
+        processed_event_payload = orjson.loads(event_payload)
+    return processed_event_payload
+
+
+async def handle_event(profile: Profile, event: EventWithMetadata):
+    """Push events from aca-py events."""
+    config_events = get_config(profile.settings).event or EventConfig.default()
+    pattern = event.metadata.pattern.pattern
+    template = config_events.event_topic_maps.get(pattern)
+
+    if not template:
+        LOGGER.warning(f"Could not infer template from pattern: {pattern}")
+        return
+
+    if "-with-state" not in template:
+        # We are only interested in state change webhooks. This avoids duplicate events
+        return
+
+    nats = profile.inject_or(NATS)
+    if not nats:
+        nats = await nats_setup(profile, event)
+
+    LOGGER.debug("Handling event: %s", event)
+    wallet_id = cast(Optional[str], profile.settings.get("wallet.id"))
+    try:
+        event_payload = process_event_payload(event.payload)
+    except TypeError:
+        try:
+            event_payload = event.payload.serialize()
+        except AttributeError:
+            try:
+                event_payload = process_event_payload(event.payload.payload)
+            except TypeError:
+                event_payload = process_event_payload(event.payload.enc_payload)
+    payload = {
+        "wallet_id": wallet_id or "base",
+        "state": event_payload.get("state"),
+        "topic": event.topic,
+        "category": _derive_category(event.topic),
+        "payload": event_payload,
+    }
+    webhook_urls = profile.settings.get("admin.webhook_urls")
+    try:
+        nats_topic = Template(template).substitute(**payload)
+        LOGGER.debug(f"Sending message {payload} with topic {nats_topic}")
+
+        origin = profile.settings.get("default_label")
+        group_id = profile.settings.get("wallet.group_id")
+
+        metadata = {"time_ns": time.time_ns()}
+        metadata_wallet_id = {"x-wallet-id": wallet_id} if wallet_id else {}
+        metadata_group_id = {"group_id": group_id} if group_id else {}
+        metadata_origin = {"origin": origin} if origin else {}
+        metadata.update(metadata_wallet_id)
+        metadata.update(metadata_group_id)
+        metadata.update(metadata_origin)
+
+        outbound = orjson.dumps({"payload": payload, "metadata": metadata})
+        await nats.publish(nats_topic, outbound)
```

Comment:
We should be using NATS jetstream to leverage the persistence mechanisms of NATS